core/src/main/java/org/mule/runtime/core/internal/routing/forkjoin/AbstractForkJoinStrategyFactory.java

```diff
@@ -91,15 +93,34 @@ public ForkJoinStrategy createForkJoinStrategy(ProcessingStrategy processingStra
                              maxConcurrency)
           .collectList()
           .doOnNext(list -> {
-            if (list.stream().anyMatch(event -> event.getError().isPresent())) {
-              throw propagate(createCompositeRoutingException(list));
+            if (list.stream().anyMatch(event -> event.getError().isPresent()
+                && !isOriginalError(event.getError().get(), original.getError()))) {
+              throw propagate(createCompositeRoutingException(list.stream()
+                  .map(event -> removeOriginalError(event, original.getError())).collect(toList())));
             }
           })
           .doOnNext(mergeVariables(original, resultBuilder))
           .map(createResultEvent(original, resultBuilder));
     };
   }
 
+  private boolean isOriginalError(Error newError, Optional<Error> originalError) {
+    if (!originalError.isPresent()) {
+      return false;
+    }
+    return originalError.get().equals(newError);
+  }
+
+  private CoreEvent removeOriginalError(CoreEvent event, Optional<Error> originalError) {
+    if (!event.getError().isPresent()) {
```

Comment:
use the optional, luke!

![image](https://user-images.githubusercontent.com/8010105/63095306-51f06800-bf41-11e9-87b9-b635c21b651b.png)

Reply:
obi wan feels lonely